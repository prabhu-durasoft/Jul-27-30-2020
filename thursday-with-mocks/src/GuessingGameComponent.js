import React from 'react';
import {GuessingGame} from './GuessingGame';

export class GuessingGameComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            game: new GuessingGame(),
            guess: 0
        }
    }
    guessChanged(e) {
        this.setState({guess: e.target.value});
    }
    guessButtonClicked() {
        const {game, guess} = this.state;
        game.play(guess);
        //Asynchronously
        this.setState({game});
    }
    render() {
        const {message, attempts} = this.state.game;
        return (<div>
            <input type="text" placeholder="Enter yourr guess" onChange={this.guessChanged.bind(this)}></input>
            <br/>
            <button onClick={this.guessButtonClicked.bind(this)}>Guess</button>
            <h1 data-test="message">{message == null ? '' : message }</h1>
            <h1 data-test="attempts">Attempts: {attempts}</h1>
        </div>);
    }
}