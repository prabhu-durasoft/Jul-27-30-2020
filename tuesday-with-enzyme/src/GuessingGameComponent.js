import React from 'react';
import {GuessingGame} from './GuessingGame';

export class GuessingGameComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            game: new GuessingGame()
        }
    }
    render() {
        return (<div></div>);
    }
}