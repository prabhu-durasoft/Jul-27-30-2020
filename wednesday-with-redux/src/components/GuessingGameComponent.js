import React from 'react';

export class GuessingGameComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            guess: 0
        }
    }
    guessChanged(e) {
        this.setState({guess: e.target.value});
    }
    guessButtonClicked() {
        this.props.play(this.state.guess);
    }
    render() {
        const {message, attempts} = this.props;
        return (<div>
            <input type="text" placeholder="Enter your guess" onChange={this.guessChanged.bind(this)}></input>
            <br/>
            <button onClick={this.guessButtonClicked.bind(this)}>Guess</button>
            <h1 data-test="message">{message == null ? '' : message }</h1>
            <h1 data-test="attempts">Attempts: {attempts}</h1>
        </div>);
    }
}