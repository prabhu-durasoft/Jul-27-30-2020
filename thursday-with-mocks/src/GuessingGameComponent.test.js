import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

import { GuessingGameComponent } from './GuessingGameComponent';
import { GuessingGame } from './GuessingGame';
Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Basic UI and state tests", () => {
    let component;
    let button;
    let attemptsHeader;
    let messageHeader;
    let guessTB;

    beforeEach(() => {
        component = shallow(<GuessingGameComponent/>);
        button  = component.find('button');
        guessTB = component.find('input[type="text"]');
        attemptsHeader = component.find('h1[data-test="attempts"]');
        messageHeader = component.find('h1[data-test="message"]');
    });

    it("should render component", () => {
        expect(component).toBeTruthy();
    });
});    