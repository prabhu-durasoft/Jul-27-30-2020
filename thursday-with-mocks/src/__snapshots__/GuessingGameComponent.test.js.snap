// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create snapshot 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Enter yourr guess"
    type="text"
  />
  <br />
  <button
    onClick={[Function]}
  >
    Guess
  </button>
  <h1
    data-test="message"
  >
    
  </h1>
  <h1
    data-test="attempts"
  >
    Attempts: 
    0
  </h1>
</div>
`;
